src/GovUk.Education.ExploreEducationStatistics.Data.Model/Repository/FootnoteRepository.cs

```diff
@@ -19,40 +18,134 @@ public FootnoteRepository(StatisticsDbContext context)
             _context = context;
         }
 
-        public IEnumerable<Footnote> GetFilteredFootnotes(
+        public async Task<Footnote> CreateFootnote(Guid releaseId,
+            string content,
+            IReadOnlyCollection<Guid> filterIds,
+            IReadOnlyCollection<Guid> filterGroupIds,
+            IReadOnlyCollection<Guid> filterItemIds,
+            IReadOnlyCollection<Guid> indicatorIds,
+            IReadOnlyCollection<Guid> subjectIds)
+        {
+            var footnote = new Footnote
+            {
+                Releases = new List<ReleaseFootnote>
+                {
+                    new()
+                    {
+                        ReleaseId = releaseId
+                    }
+                },
+                Content = content,
+                Filters = filterIds.Select(filterId => new FilterFootnote
+                {
+                    FilterId = filterId
+                }).ToList(),
+                FilterGroups = filterGroupIds.Select(filterGroupId => new FilterGroupFootnote
+                {
+                    FilterGroupId = filterGroupId
+                }).ToList(),
+                FilterItems = filterItemIds.Select(filterItemId => new FilterItemFootnote
+                {
+                    FilterItemId = filterItemId
+                }).ToList(),
+                Indicators = indicatorIds.Select(indicatorId => new IndicatorFootnote
+                {
+                    IndicatorId = indicatorId
+                }).ToList(),
+                Subjects = subjectIds.Select(subjectId => new SubjectFootnote
+                {
+                    SubjectId = subjectId
+                }).ToList()
```

Comment:
Seems like it's possible for the code to attempt to insert multiple links with the same ids i.e. dupes.

I'm guessing this would get picked up by the DB when it tries to create the links, but I'd be tempted to ensure that we don't get into this situation by filtering out any dupe ids.

We could either do this by calling `Distinct().Select(...)` on each link, or change the method parameters to use `IReadOnlySet<Guid>` (the consuming code would have to create the sets beforehand).

Reply:
Decided to go with `IReadOnlySet<Guid>` all the way up to the request types `FootnoteCreateRequest`  and `FootnoteUpdateRequest`. Thanks.

Reply:
By the way, I've added tests now for creating footnotes, not in `FootnoteRepositoryTest` but testing it indirectly through `FootnoteServiceTests` in the Footnotes Ordering PR - https://github.com/dfe-analytical-services/explore-education-statistics/pull/3713/files#diff-1959d3889e4233e87d3fbd3aec8c7e32b54b035b3d563467c05c29adcf040283